p2p/raintree/network_test.go

```diff
@@ -11,69 +11,121 @@ import (
-	numAddressesInAddressBook := 3
-	addrBook := getAddrBook(nil, numAddressesInAddressBook)
+	// Start with a peerstore which contains self and some number of peers: the
+	// initial value of `expectedPStoreSize`.
+	expectedPStoreSize := 3
+	pstore := getPeerstore(nil, expectedPStoreSize)
+
 	selfAddr, err := cryptoPocket.GenerateAddress()
 	require.NoError(t, err)
 	selfPeer := &typesP2P.NetworkPeer{Address: selfAddr}
-	addrBook = append(addrBook, &typesP2P.NetworkPeer{Address: selfAddr})
+
+	// TODO _THIS_COMMIT: finish comment
+	// Add self to peerstore because ... it's expected by something
+	err = pstore.AddPeer(&typesP2P.NetworkPeer{Address: selfAddr})
+	require.NoError(t, err)
+	expectedPStoreSize++
 
 	busMock := mockBus(ctrl)
-	addrBookProviderMock := mockAddrBookProvider(ctrl, addrBook)
+	peerstoreProviderMock := mockPeerstoreProvider(ctrl, pstore)
 	currentHeightProviderMock := mockCurrentHeightProvider(ctrl, 0)
 
-	network := NewRainTreeNetwork(selfAddr, busMock, addrBookProviderMock, currentHeightProviderMock).(*rainTreeNetwork)
-	stateView := network.peersManager.GetPeersView()
-	require.Equal(t, numAddressesInAddressBook+1, len(stateView.GetAddrs())) // +1 to account for self in the addrBook as well
-
-	// removing a peer
-	peer := addrBook[1]
-	err = network.RemovePeerFromAddrBook(peer)
+	network := NewRainTreeNetwork(selfAddr, busMock, peerstoreProviderMock, currentHeightProviderMock).(*rainTreeNetwork)
+
+	// Ensure expected starting size / lengths are consistent.
+	peersView, peerAddrs, peers := getPeersViewParts(network.peersManager)
+	require.Equal(t, expectedPStoreSize, pstore.Size())
+	require.Equal(t, expectedPStoreSize, len(peerAddrs))
+	require.Equal(t, expectedPStoreSize, len(peers))
+
+	var peerToRemove sharedP2P.Peer
+	// Ensure we don't remove selfPeer. `Peerstore` interface isn't aware
+	// of the concept of "self" so we have to find it.
+	for _, peer := range pstore.GetAllPeers() {
+		if peer.GetAddress().Equals(selfAddr) {
+			continue
+		}
+		peerToRemove = peer
+		break
+	}
+	require.NotNil(t, peerToRemove, "did not find selfAddr in peerstore")
+
+	// Remove peerToRemove
+	err = network.RemovePeer(peerToRemove)
 	require.NoError(t, err)
+	expectedPStoreSize--
+
+	peersView, peerAddrs, peers = getPeersViewParts(network.peersManager)
+	removedAddr := peerToRemove.GetAddress()
+	getPeer := func(addr cryptoPocket.Address) sharedP2P.Peer {
+		return peersView.GetPeerstore().GetPeer(addr)
+	}
+
+	// Ensure updated size / lengths are consistent.
+	require.Equal(t, expectedPStoreSize, pstore.Size())
+	require.Equal(t, expectedPStoreSize, len(peerAddrs))
+	require.Equal(t, expectedPStoreSize, len(peers))
+
+	require.Equal(t, selfPeer, getPeer(selfAddr), "Peerstore does not contain self")
+	require.Nil(t, getPeer(removedAddr), "Peerstore contains removed peer")
+}
+
+func getPeersViewParts(pm sharedP2P.PeerManager) (
+	view sharedP2P.PeersView,
+	addrs []string,
+	peers sharedP2P.PeerList) {
```

Comment:
NIT: Move `) {` to the next line.

Reply:
Nice catch, thanks! :raised_hands:  I'm not sure what happened there.